ci_framework/plugins/modules/tempest_list_allowed.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2023, Arx Cruz <acruz@redhat.com>
+# Apache License Version 2.0 (see LICENSE)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: tempest_list_allowed
+
+short_description: Parse filtered tests to tempest
+
+description:
+  - Parse filtered tests to be executed by tempest
+
+options:
+  yaml_file:
+    description:
+    - Path to a yaml file containing the tests in openstack-tempest-skiplist
+      format
+    required: True
+    type: str
+  job:
+    description:
+    - Name of the job to be used in the filter.
+    required: False
+    type: str
+  groups:
+    description:
+    - List of groups to be used in the filter. It has more precedence than job
+    required: False
+    type: list
+    elements: str
+  release:
+    description:
+    - Release version
+    type: str
+    default: master
+
+author:
+  - Arx Cruz (@arxcruz)
+
+'''
+
+EXAMPLES = r'''
+- name: Get list of allowed tests
+  tempest_list_allowed:
+    yaml_file: /tmp/allowed.yaml
+    job: tripleo-ci-centos-8-standalone
+    groups:
+     - default
+'''
+
+RETURN = r'''
+allowed_tests:
+  description:
+    - List of tests filtered
+  returned: Always
+  type: list
+  sample: [
+    "tempest.api.volume.test_volumes_snapshots.VolumesSnapshotTestJSON"
+    ]
+'''
+
+
+from ansible.module_utils.basic import AnsibleModule
+import yaml
+
+
+def _filter_allowed_tests(groups, release, job, yaml_file):
+    tests = []
+    if len(yaml_file) > 0:
```

Comment:
List and dicts in python evaluates to false in python, no need to call len. In fact that's the pyhonic way of checking if it's empty.